assignment-1/Program.cs

```diff
@@ -0,0 +1,274 @@
+        Console.WriteLine("Welcome to the Library Management System");
+        while (!exit)
+        {
+            Console.WriteLine("\n1. Create Admin Account");
+            Console.WriteLine("2. Add Book");
+            Console.WriteLine("3. Register User");
+            Console.WriteLine("4. Issue Book");
+            Console.WriteLine("5. Return Book");
+            Console.WriteLine("6. List books issued by User");
+            Console.WriteLine("7. Get book inventory info");
+            Console.WriteLine("8. List users with the book");
+            Console.WriteLine("9. Calculate fine for user");
+            Console.WriteLine("10. Manage Inventory (Increase)");
+            Console.WriteLine("11. Manage Inventory (Reduce)");
+            Console.WriteLine("12. Exit\n");
+
+            Console.Write("Please select an option: ");
+            string choice = Console.ReadLine();
+
+            switch (choice)
+            {
+                case "1":
+                    CreateAdmin(library);
+                    break;
+                case "2":
+                    AddBook(library);
+                    break;
+                case "3":
+                    RegisterUser(library);
+                    break;
+                case "4":
+                    IssueBook(library);
+                    break;
+                case "5":
+                    ReturnBook(library);
+                    break;
+                case "6":
+                    ListUserBooks(library);
+                    break;
+                case "7":
+                    GetBookInventoryInfo(library);
+                    break;
+                case "8":
+                    ListIssuersOfBook(library);
+                    break;
+                case "9":
+                    CalculateFine(library);
+                    break;
+                case "10":
+                    ManageInventory(library);
+                    break;
+                case "11":
+                    ReduceInventory(library);
+                    break;
+                case "12":
+                    exit = true;
+                    break;
+                default:
+                    Console.WriteLine("Invalid option. Please try again.");
+                    break;
+            }
+        }
+    }
+
+    static void CreateAdmin(Library library)
+    {
+        string username;
+        do
+        {
+            Console.Write("\nEnter admin username: ");
+            username = Console.ReadLine();
+
+            if (string.IsNullOrWhiteSpace(username))
+            {
+                Console.WriteLine("Username cannot be empty.");
+            }
+        } while (string.IsNullOrWhiteSpace(username));
+
+        Console.Write("Enter admin password: ");
+        string password = Console.ReadLine();
+
+        library.CreateAndAddAdmin(username, password);
+
+        Console.WriteLine("\nAdmin account created successfully.");
+    }
+
+
+    static void RegisterUser(Library library)

```

Comment:
All these function interacts with the User directly so you could have moved these functions to `Views` folder. There can be multiple views, AdminView, StudentView and TeacherView and then there can be some common views which would be inherited by all these Views like `ListAllBooks`